### PR TITLE
NCO-476: add debian packaging

### PR DESCRIPTION
**Pull request by pgmillon**
* Debian packaging using jdeb
* Systemd unit
* Ubuntu Vagrantfile for testing (not automated)

### CODE REVIEW

File: src/main/resources/aws-smtp-relay.service [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# /usr/lib/systemd/system/tor.service
```
Comment (loopingz) [inline src/main/resources/aws-smtp-relay.service]: this is the result of a copy/paste we should update

File: src/main/resources/aws-smtp-relay.service [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+[Unit]
+Description=AWS SMTP Relay
+After=network.target
+
+[Service]
+User=aws-smtp-relay
+Type=simple
+ExecStart=/usr/bin/java -jar /usr/share/aws-smtp-relay/aws-smtp-relay-${project.version}-jar-with-dependencies.jar -b 0.0.0.0
```
Comment (loopingz) [inline src/main/resources/aws-smtp-relay.service]: by default the SMTP should listen to 127.0.0.1 to avoid any security risk

**Review by loopingz — CHANGES_REQUESTED** [review]
What is the purpose on Vagrant in this PR?

**Comment by pgmillon** [comment]
The Vagrantfile was because I intended to add integration tests of the SystemD unit.
Unfortunately the only project I found has been discontinued (https://github.com/nicoulaj/vagrant-maven-plugin).
I'll remove it.

File: src/main/resources/aws-smtp-relay.service [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# /usr/lib/systemd/system/tor.service
+[Unit]
+Description=AWS SMTP Relay
+After=network.target
+
+[Service]
```
Comment (mikepii) [inline src/main/resources/aws-smtp-relay.service:6]: How about `SuccessExitStatus=130 143`? I believe Java will exit with one of those codes on success and this is needed so the unit doesn't enter a "failed" state

Comment (pgmillon) [reply src/main/resources/aws-smtp-relay.service:6]: Agreed, fixed

File: src/main/resources/aws-smtp-relay.service (hunk 2 above)
Comment (pgmillon) [reply src/main/resources/aws-smtp-relay.service]: Fixed

File: src/main/resources/aws-smtp-relay.service (hunk 1 above)
Comment (pgmillon) [reply src/main/resources/aws-smtp-relay.service]: Fixed

**Comment by pgmillon** [comment]
Vagrantfile removed.
Also, manage both upstart and systemd inspired on #23

File: src/main/resources/aws-smtp-relay.service [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# /usr/lib/systemd/system/aws-smtp-relay.service
+[Unit]
+Description=AWS SMTP Relay
+After=network.target
+
+[Service]
+User=aws-smtp-relay
+Type=simple
+ExecStart=/usr/bin/java -jar /usr/share/aws-smtp-relay/aws-smtp-relay-${project.version}-jar-with-dependencies.jar -b 127.0.0.1
```
Comment (loopingz) [inline src/main/resources/aws-smtp-relay.service]: Thanks for all the current changes, i would say the last one would be to make the bining configurable in a /etc/ file with a default of 127.0.0.1

**Comment by pgmillon** [comment]
- Review upstart init
- Add init configuration

**Review by loopingz — APPROVED** [review]

**Pull request merged**